manifests/instance.pp

```diff
@@ -228,32 +234,49 @@
       require => Exec["Setup ${title} DS"]
     }
 
-    ensure_resource('service', "dirsrv@${title}",
-      {
-        ensure     => 'running',
-        enable     => true,
-        hasrestart => true
-      }
-    )
+    ensure_resource('ds389::instance::service', $title)
 
-    ds389::config::item { "Set nsslapd-listenhost on ${title}":
-      key          => 'nsslapd-listenhost',
-      value        => $listen_address,
-      admin_dn     => $root_dn,
-      pw_file      => $_ds_pw_file,
-      host         => $listen_address,
-      port         => $port,
-      service_name => $title
+    # This needs to happen first so that we can skip through to ldapi afterwards
+    ds389::instance::attr::set { "Configure LDAPI for ${title}":
+      instance_name => $title,
+      attrs         => {
+        'cn=config' => {
+          'nsslapd-ldapilisten' => 'on',
+          'nsslapd-localssf'    => 99999
```

Comment:
Seems like we would not want to use the default value for all instances.

Reply:
Looks like the logic in set.pp assumes a path with the instance name if it is not set, but that path is not the RHEL default of `/var/run/dirsrv/slapd-example.socket`.  I'm missing something.

Reply:
1. `ldapifilepath` has a sane default and is discovered via the `ds389__instances` fact
2. You definitely want 99999 on all instances since anything lower than `minssf` won't allow you to access the system (and what's the point of that for a local socket interface?)
3. This is just a sane default if there isn't something in `ds389__instances` and you're trying to force it. It might not work (but it probably will).